spec/usage/extend/custom_component_spec.rb

```diff
@@ -0,0 +1,169 @@
+      get '/custom_component_test', to: 'component_test#my_action'
+    end
+    Rails.application.reload_routes!
+
+  end
+
+  it 'by orchestrating existing core components' do
+
+    module Components end
+
+    class Components::CrazyComponent < Matestack::Ui::StaticComponent
+
+      def response
+        components {
+          div id: 'my-component-1' do
+            plain "I'm a static component!"
+          end
+        }
+      end
+
+    end
+
+    class Pages::ExamplePage < Matestack::Ui::Page
+
+      def response
+        components {
+          div id: 'div-on-page' do
+            custom_crazyComponent
+          end
+        }
+      end
+
+    end
+
+    visit '/custom_component_test'
+
+    expect(page).to have_xpath('//div[@id="div-on-page"]/div[@id="my-component-1" and contains(.,"I\'m a static component!")]')
+
+  end
+
+  it 'by extending actionview components - static' do
+
+    module Components end
+
+    class Components::TimeAgo < Matestack::Ui::StaticActionviewComponent
+    # class Components::TimeAgo < Matestack::Ui::Core::Actionview::Static # without alias
+
+      def prepare
+        @from_time = Time.now - 3.days - 14.minutes - 25.seconds
+      end
+
+      def response
+        components {
+          div id: 'my-component-1' do
+            plain time_ago_in_words(@from_time)
+          end
+        }
+      end
+
+    end
+
+    class Pages::ExamplePage < Matestack::Ui::Page
+
+      def response
+        components {
+          div id: 'div-on-page' do
+            custom_timeAgo
+          end
+        }
+      end
+
+    end
+
+    visit '/custom_component_test'
+
+    expect(page).to have_xpath('//div[@id="div-on-page"]/div[@id="my-component-1" and contains(.,"3 days")]')
+
+  end
+
+  it 'by extending actionview components - dynamic' do
+
+    module Components end
+
+    class Components::TimeClick < Matestack::Ui::DynamicActionviewComponent
+      # class Components::TimeClick < Matestack::Ui::Core::Actionview::Dynamic # without alias
+
+      def prepare
+        @start_time = Time.now
+      end
+
+      def response
+        components {
+          div id: 'my-component-1' do
+            plain "{{dynamic_value}} #{distance_of_time_in_words(@start_time, Time.now, include_seconds: true)}"
+          end
+        }
+      end
+
+    end
+
+    component_definition = <<~javascript
+
+      MatestackUiCore.Vue.component('custom-timeclick', {
+        mixins: [MatestackUiCore.componentMixin],
+        data: function data() {
+          return {
+            dynamic_value: "Now I show: "
+          };
+        },
+        mounted(){
+          const self = this
+          setTimeout(function () {
+            self.dynamic_value = "Later I show: "
+          }, 300);
+        }
+      });
+
+    javascript
+
+    class Pages::ExamplePage < Matestack::Ui::Page
+
+      def response
+        components {
+          div id: 'div-on-page' do
+            async rerender_on: "refresh" do
```

Comment:
adding a comment here, why a rerender is necessary, would be important. Something like: "The `async rerender` is only used in this test because we add the vue.js component definition during runtime and need to reinit this DOM-part in order to trigger vue.js to mount the component properly."
Never use this example in a documentation as it would be completely misleading ;)

Reply:
Good point, added the explanation!